src/neural/network_blas.cc

```diff
@@ -220,23 +223,51 @@ class BlasNetwork : public Network {
       Transforms::InvertBatchNormStddev(conv2.bn_stddivs);
     }
 
-    Transforms::OffsetBatchNormMeans(weights_.policy.bn_means, weights_.policy.biases);
+    Transforms::OffsetBatchNormMeans(weights_.policy.bn_means,
+                                     weights_.policy.biases);
     Transforms::InvertBatchNormStddev(weights_.policy.bn_stddivs);
 
-    Transforms::OffsetBatchNormMeans(weights_.value.bn_means, weights_.value.biases);
+    Transforms::OffsetBatchNormMeans(weights_.value.bn_means,
+                                     weights_.value.biases);
     Transforms::InvertBatchNormStddev(weights_.value.bn_stddivs);
 
 #ifdef USE_OPENBLAS
-// openblas_set_num_threads(1);
-// printf("BLAS Core: %s\n", openblas_get_corename());
+    int num_procs = openblas_get_num_procs();
+    blas_cores = std::min(num_procs, blas_cores);
+    openblas_set_num_threads(blas_cores);
+    if (verbose) {
+      const char* core_name = openblas_get_corename();
+      const char* config = openblas_get_config();
+      printf("BLAS vendor: OpenBlas.\n");
```

Comment:
Please output that to stderr rather than stdout.
If it's during initialization only, it's fine to remove "verbose" parameter and output always.

Also consider using C++ `std::cerr` instead. (but I don't insist)

Reply:
Yes, that has to be discussed. 
I have no opinion of stdout/stderr for initial verbose comments...
But if we decide something, let use change all the backends at the same time.

Reply:
sorry did get you there, "cleanup the printf laying around"

Reply:
do `rgrep printf`, and replace everything it finds with standard C++ io

Reply:
I really dislike C++ io which is an abomination imho and I would stick with printf(stderr, ...)